src/components/v2/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { Suspense, useEffect, useState } from 'react';
+import { Switch, Route } from 'react-router-dom';
+import { URLS } from '../../config';
+import { useRouteMatch, useParams, Redirect } from 'react-router';
+import { DetailsProgressing } from '../common';
+import './lib/bootstrap-grid.min.css';
+import ValuesComponent from './values/ChartValues.component';
+import AppHeaderComponent from './headers/AppHeader.component';
+import ChartHeaderComponent from './headers/ChartHeader.component';
+import { getInstalledAppDetail } from '../charts/charts.service';
+import { getInstalledChartDetail } from './appDetails/appDetails.api';
+import AppDetailsComponent from './appDetails/AppDetails.component';
+import { EnvType } from './appDetails/appDetails.type';
+import IndexStore from './appDetails/index.store';
+
+let initTimer = null;
+
+function RouterComponent({ envType }) {
+    const [isLoading, setIsLoading] = useState(true);
+    const params = useParams<{ appId: string; envId: string; nodeType: string }>();
+    const { path } = useRouteMatch();
+
+    useEffect(() => {
+        IndexStore.setEnvDetails(envType, +params.appId, +params.envId);
+
+        setIsLoading(true);
+
+        if (params.appId && params.envId) {
+            if (initTimer) {
+                clearTimeout(initTimer);
+            }
+
+            init();
+        }
+    }, [params.appId, params.envId]);
+
+    // clearing the timer on component unmount
+    useEffect(() => {
+        return (): void => {
+            if (initTimer) {
+                clearTimeout(initTimer);
+            }
+        };
+    }, []);
+
+    const init = async () => {
+        try {
+            let response = null;
+
+            if (envType === EnvType.CHART) {
+                response = await getInstalledChartDetail(+params.appId, +params.envId);
+            } else {
+                response = await getInstalledAppDetail(+params.appId, +params.envId);
```

Comment:
should have been used from src/components/v2/appDetails/appDetails.api.ts instead of src/components/charts/charts.service.ts

Reply:
also we need to handle the case when env is not in the url,  in that case we want user to select the environment, currently loader doesnt stop

Reply:
This behaviour is only true in case of devtron app. (Not in the scope of Charts)